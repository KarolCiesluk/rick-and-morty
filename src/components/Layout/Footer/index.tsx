import { StyledFooter } from "./styled";

interface FooterProps {
  footerText: string;
}

const Footer = ({ footerText }: FooterProps) => (
  <StyledFooter>
    <span>{footerText}</span>
  </StyledFooter>
);

export default Footer;