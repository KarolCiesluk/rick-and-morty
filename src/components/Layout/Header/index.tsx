import { StyledHeader } from "./styled";

interface HeaderProps {
  children: React.ReactNode;
}

const Header = ({ children }: HeaderProps) => (
  <StyledHeader>
    <h1>{children}</h1>
  </StyledHeader>
);

export default Header;