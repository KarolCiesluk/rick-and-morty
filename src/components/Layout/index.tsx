import Footer from "./Footer";
import Header from "./Header";

interface LayoutProps {
  children: React.ReactNode;
}

const Layout = ({ children }: LayoutProps) => (
  <div>
    <Header>Rick and Morty app</Header>
    <main>
      {children}
    </main>
    <Footer footerText="© 2022 | Coded by Karol Cieśluk" />
  </div>
);

export default Layout;