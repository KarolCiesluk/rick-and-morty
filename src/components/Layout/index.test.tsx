import renderer from "react-test-renderer";

import Layout from ".";

it("renders ok", () => {
  const tree = renderer
    .create(<Layout><div>Snapshot test</div></Layout>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});