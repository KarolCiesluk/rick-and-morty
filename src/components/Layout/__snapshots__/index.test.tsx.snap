// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ok 1`] = `
<div
  className="sc-dkPtRN kcwarw"
>
  <header
    className="sc-gsDKAQ eGePnG"
  >
    <h1>
      Rick and Morty app
    </h1>
  </header>
  <main>
    <div>
      Snapshot test
    </div>
  </main>
  <footer
    className="sc-bdvvtL boMJDT"
  >
    <span>
      © 2022 | Coded by Karol Cieśluk
    </span>
  </footer>
</div>
`;
